web/js/codeworld_shared.js

```diff
@@ -122,13 +122,111 @@ var hintBlacklist = [
     "Pi",
 ];
 
-var codeWorldBuiltinsDocs = {};
+
+// codeWorldDocs is variable containing annotations and documentation
+// of builtin and user-defined variables.
+// Expected format:
+// codeWorldDocs = {
+//   codeWorldLogo: {
+//     annotation: "Picture",
+//     doc: "The CodeWorld logo."
+//   }
+// }
+
+var codeWorldDocs = {};
+
+function renderHover(keyword) {
```

Comment:
This looks like it only handles lines of the form `foo :: Type` or `foo`.  What about `(*#^) :: Type`, or `data Foo`, or `type Foo = Bar`?

Reply:
Will extend rules.